ui/components/AlertsTable.tsx

```diff
@@ -1,19 +1,35 @@
+import qs from "query-string";
 import * as React from "react";
 import styled from "styled-components";
 import { useFeatureFlags } from "../hooks/featureflags";
-import { Alert, CrossNamespaceObjectRef } from "../lib/objects";
+import { Alert, CrossNamespaceObjectRef, Kind } from "../lib/objects";
+import { V2Routes } from "../lib/types";
 import { statusSortHelper } from "../lib/utils";
 import DataTable, {
   Field,
   filterByStatusCallback,
   filterConfig,
 } from "./DataTable";
+import { filterSeparator } from "./FilterDialog";
 import KubeStatusIndicator from "./KubeStatusIndicator";
+import Link from "./Link";
 type Props = {
   className?: string;
   rows?: Alert[];
 };
 
+export const makeEventSourceLink = (obj: CrossNamespaceObjectRef) => {
```

Comment:
Can you please add tests for this function? It's just asking for it. And I think the function return type should be specified explicitly.

Reply:
I started writing some tests, and then I was essentially saying to expect the output to be exactly what the return value of the function is, just with the arguments subbed in. Is there a better way to test it than just plugging in a fake obj and expecting the return value?

Reply:
Normally, yes, you just provide fake objects with different fields, write expected values (the returned string values that you know should be correct for this object) first and then check if what your function produces matched the expected values (of course you can first run the function and copy/paste the actual values to expected but the former is considered "the right way" ™). Tests also serve as documentation and for catching regressions so they should be useful even if they look simple.

Reply:
Tests added under `AlertsTable.test.tsx`!!